packages/block-editor/src/components/block-list/use-block-props/use-scroll-into-view.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { useReducedMotion, useRefEffect } from '@wordpress/compose';
+
+export function useScrollIntoView( { isSelected } ) {
+	const prefersReducedMotion = useReducedMotion();
+	return useRefEffect(
+		( node ) => {
+			if ( isSelected ) {
+				const { ownerDocument } = node;
+				const { defaultView } = ownerDocument;
+				if ( ! defaultView.IntersectionObserver ) {
+					return;
+				}
+				const observer = new defaultView.IntersectionObserver(
```

Comment:
Do we really need an intersection observer here? Can't we just scroll into view directly? Or check the position at the moment of the block selection. An observer is something that runs constantly.

Reply:
I know this calls disconnect on first observe but still.

Reply:
It disconnects immediately, the observer will always immediately call the callback with the initial intersecting state of the element. As for as I know this is the only way to check if an element is visible without using a library.

> The Intersection Observer API allows you to configure a callback that is called when either of these circumstances occur:
>
> * A target element intersects either the device's viewport or a specified element. That specified element is called the root element or root for the purposes of the Intersection Observer API.
> * The first time the observer is initially asked to watch a target element.

https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#intersection_observer_concepts_and_usage